pydicom/pixel_data_handlers/util.py

```diff
@@ -1,39 +1,415 @@
 # Copyright 2008-2018 pydicom authors. See LICENSE file for details.
 """Utility functions used in the pixel data handlers."""
 
-import sys
+from sys import byteorder
 
-sys_is_little_endian = (sys.byteorder == 'little')
+try:
+    import numpy as np
+    HAVE_NP = True
+except ImportError:
+    HAVE_NP = False
 
 
-def dtype_corrected_for_endianess(is_little_endian, numpy_dtype):
-    """Adapts the given numpy data type for changing the endianess of the
-    dataset, if needed.
+def convert_colour_space(arr, current, desired):
+    """Convert the image(s) in `arr` from one colour space to another.
 
-        Parameters
-        ----------
-        is_little_endian : bool
-            The endianess of the affected dataset.
-        numpy_dtype : numpy.dtype
-            The numpy data type used for the pixel data without considering
-            endianess.
+    Parameters
+    ----------
+    arr : numpy.ndarray
+        The image(s) as an ndarray with shape (frames, rows, columns, planes)
+        or (rows, columns, planes).
+    current : str
+        The current colour space, should be a valid value for (0028,0004)
+        *Photometric Interpretation*. One of 'RGB', 'YBR_FULL'.
+    desired : str
+        The desired colour space, should be a valid value for (0028,0004)
+        *Photometric Interpretation*. One of 'RGB', 'YBR_FULL'.
 
-        Raises
-        ------
-        ValueError
-            If `is_little_endian` id None, e.g. not initialized.
+    Returns
+    -------
+    numpy.ndarray
+        The image(s) converted to the desired colour space.
+    """
+    if not HAVE_NP:
+        raise ImportError(
+            "Numpy is required to convert the color space."
+        )
+
+    # No change needed
+    if current == desired:
+        return arr
+
+    _converters = {
+        'YBR_FULL': {
+            'RGB': _convert_YBR_FULL_to_RGB
+        },
+        'RGB': {
+            'YBR_FULL': _convert_RGB_to_YBR_FULL,
+        }
+    }
+    try:
+        converter = _converters[current][desired]
+    except KeyError:
+        raise NotImplementedError(
+            "Conversion from {0} to {1} is not supported."
+            .format(current, desired)
+        )
+
+    return converter(arr)
+
+
+convert_color_space = convert_colour_space
```

Comment:
Ah, that partly answers my question above... though I think we should just stick to one style.